core/handlers/v2/diff_handler.go

```diff
@@ -59,6 +58,42 @@ func (this *DiffHandler) Delete(w http.ResponseWriter, req *http.Request, next h
 	handlers.WriteResponse(w, []byte(""))
 }
 
+func (this *DiffHandler) GetFilteredData(w http.ResponseWriter, req *http.Request, next http.HandlerFunc) {
+	requestBody, err := ioutil.ReadAll(req.Body)
+	if err != nil {
+		handlers.WriteErrorResponse(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+	var diffFilterView DiffFilterView
+	err = json.Unmarshal(requestBody, &diffFilterView)
+	if err != nil {
+		handlers.WriteErrorResponse(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	diffsToReturn := ConvertToResponseDiffView(this.Hoverfly.GetFilteredDiff(diffFilterView))
+	marshal, err := json.Marshal(DiffView{
+		Diff: diffsToReturn,
+	})
+	if err != nil {
+		handlers.WriteErrorResponse(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+	handlers.WriteResponse(w, marshal)
+}
+
+func ConvertToResponseDiffView(responsesDiff map[SimpleRequestDefinitionView][]DiffReport) []ResponseDiffForRequestView {
```

Comment:
should keep it unexported `convertToResponseDiffView`

Reply:
yea, done. Also added UTs and documentation.